apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx

```diff
@@ -1,78 +1,66 @@
-import { NavigationEvents, NavigationScreenProps } from 'react-navigation';
-import HeaderButtons from 'react-navigation-header-buttons';
-import { Ionicons } from '@expo/vector-icons';
 
+import HeaderIconButton, { HeaderContainerRight } from '../../components/HeaderIconButton';
 import ContactsList from './ContactsList';
 import * as ContactUtils from './ContactUtils';
 
-const CONTACT_PAGE_SIZE = 500;
+// type StackParams = {
+//   ContactDetail: { id: string };
+// };
 
-interface State {
-  contacts: Contacts.Contact[];
-  hasPreviousPage: boolean;
-  hasNextPage: boolean;
-  permission?: boolean;
-  refreshing: boolean;
-}
-
-export default class ContactsScreen extends React.Component<NavigationScreenProps, State> {
-  static navigationOptions = () => {
-    return {
-      title: 'Contacts',
-      headerRight: (
-        <HeaderButtons
-          IconComponent={Ionicons}
-          OverflowIcon={<Ionicons name="ios-more" size={23} color="blue" />}
-          iconSize={23}
-          color="blue"
-        >
-          <HeaderButtons.Item
-            title="add"
-            iconName="md-add"
-            onPress={async () => {
-              const randomContact = { note: 'Likes expo...' };
-
-              // @ts-ignore
-              ContactUtils.presentNewContactFormAsync({ contact: randomContact });
-              // ContactUtils.presentUnknownContactFormAsync({
-              //   contact: randomContact,
-              // });
-            }}
-          />
-        </HeaderButtons>
-      ),
-    };
-  }
+const CONTACT_PAGE_SIZE = 500;
 
-  _rawContacts: { [contactId: string]: Contacts.Contact } = {};
-  readonly state: State = {
-    contacts: [],
-    hasPreviousPage: false,
-    hasNextPage: false,
-    refreshing: false,
-  };
+export default function ContactsScreen({
+  navigation,
+}: {
+  navigation: any /* StackNavigationProp<StackParams>; */;
+}) {
+  const [permission] = usePermissions(Permissions.CONTACTS, { ask: true });
 
-  componentDidFocus = async () => {
-    await this.checkPermissionAsync();
-    await this.loadAsync();
+  if (!permission) {
+    return (
+      <View style={styles.permissionContainer}>
+        <Text>No Contact Permission</Text>
+      </View>
+    );
   }
 
-  checkPermissionAsync = async () => {
-    const { status } = await Permissions.askAsync(Permissions.CONTACTS);
-    this.setState({ permission: status === 'granted' });
-  }
+  return <ContactsView navigation={navigation} />;
+}
 
-  loadAsync = async ({ distanceFromEnd }: { distanceFromEnd?: number } = {}, restart = false) => {
-    if (!this.state.permission || this.state.refreshing) {
+function ContactsView({ navigation }: { navigation: any /* StackNavigationProp<StackParams> */ }) {
+  let _rawContacts: Record<string, Contacts.Contact> = {};
+
+  const [contacts, setContacts] = React.useState<Contacts.Contact[]>([]);
+  const [hasPreviousPage, setHasPreviousPage] = React.useState<boolean>(true);
```

Comment:
The inferred type should be sufficient:
```suggestion
  const [hasPreviousPage, setHasPreviousPage] = React.useState(true);
```